tool/main.re

```diff
@@ -246,22 +250,143 @@ static MULTIPLAYER_MENU = Ui::new("Multiplayer:", List::of(
+}
+
 static UTIL_MENU = Ui::new("Util:", List::of(
     UiElement::Input(Input {
-        label: Text { text: "Save Recording" },
-        input: "",
+        label: Text { text: "Player Name" },
+        input: PLAYER_NAME,
         onclick: fn(input: string) {
-            tas_save_recording(input);
+            SETTINGS.set_multiplayer_name(input);
         },
         onchange: fn(input: string) {},
     }),
-    UiElement::Input(Input {
+    UiElement::Button(UiButton {
+        label: Text { text: "Save Recording" },
+        onclick: fn(label: Text) {
+            handle_operation(ReplayButtonOp::Save);
+        }
+    }),
+    UiElement::Button(UiButton {
         label: Text { text: "Load Recording" },
-        input: "",
-        onclick: fn(input: string) {
-            tas_load_recording(input);
-        },
-        onchange: fn(input: string) {},
+        onclick: fn(label: Text) {
+            handle_operation(ReplayButtonOp::Load);
+        }
+    }),
+    UiElement::Button(UiButton {
+        label: Text { text: "Delete Recording" },
+        onclick: fn(label: Text) {
+            RECORDING_NAME_LABEL.text = f"Recording name";
+            let recordings_list = Tas::list_recordings();
+            let mut recordings = List::of(
+                UiElement::Button(UiButton {
+                    label: Text { text: "Back" },
+                    onclick: fn(label: Text) { leave_ui() },
+                }),
+                UiElement::Input(Input {
+                    label: RECORDING_NAME_LABEL,
+                    input: "",
+                    onclick: fn(input: string) {
+                        let recordings_list = Tas::list_recordings();
+                        if input.len_utf8() == 0 {
+                            RECORDING_NAME_LABEL.text = f"Recording name (Error: empty name)";
+                            return;
+                        }
+                        if recordings_list.contains(input) {
+                            RECORDING_NAME_LABEL.text = f"Recording name";
+                            Tas::remove_recording(input);
+                        } else {
+                            RECORDING_NAME_LABEL.text = f"Recording name (Error: no such recording)";
+                        }
+                    },
+                    onchange: fn(input: string) {}
+                }),
+            );
+            for recording in recordings_list {
+                recordings.push(UiElement::Button(UiButton {
+                    label: Text { text: recording },
+                    onclick: fn(label: Text) {
+                        Tas::remove_recording(label.text); // The list can only update when you re-enter the menu
+                        RECORDING_NAME_LABEL.text = f"Recording name (Deleted {label.text})";
+                    },
+                }));
+            }
+            enter_ui(Ui::new("Recording Options:", recordings));
+        }
     }),
```

Comment:
Why is this still here?

Reply:
LMFAO I didn't realize this was still in here